### ISSUE TITLE
"resizable" set to false is not disabling the maximize button in iced 0.10.0 but was working in earlier versions

### ISSUE THREAD

**Issue by mayankjoshii** [iced-rs/iced#2121]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

The `resizable` data member of the struct [iced](https://docs.rs/iced/latest/iced/index.html)::[window](https://docs.rs/iced/latest/iced/window/index.html)::[Settings](https://docs.rs/iced/latest/iced/window/struct.Settings.html#) struct when set to` false` is not disabling the minimize button in the menu bar in the latest **iced 0.10.0** version. 

**Success Case:** 
It was working as expected in the older versions of iced till **iced 0.9.0**. 

This the screen shot of progress example built with iced 0.9.0. Here the maximize button is disable when the `resizable` was set to `false`:

![iced-9](https://github.com/iced-rs/iced/assets/98211866/ab464d38-eebf-4db3-b490-2ab9897c250f)

**Failure Case:**
This the screen shot of progress example built with iced 0.10.0. Here the maximize button is still enabled even if the `resizable` is set to `false`:

![iced-10](https://github.com/iced-rs/iced/assets/98211866/2585c137-3886-401e-bf5c-63ee4d5e0863)

Here's the code snippet to reproduce and test the behavior: 

```
use iced::window;
use iced::settings::Settings;
use iced::{Element, Sandbox};
use iced::widget::{column, progress_bar, slider};

pub fn main() -> iced::Result {
    Progress::run(Settings{
	     window: iced::window::Settings 
	     { 
                    resizable: false,
                    ..window::Settings::default()
             },
             ..Settings::default()
         }
    )
}

#[derive(Default)]
struct Progress {
    value: f32,
}

#[derive(Debug, Clone, Copy)]
enum Message {
    SliderChanged(f32),
}

impl Sandbox for Progress {
    type Message = Message;

    fn new() -> Self {
        Self::default()
    }

    fn title(&self) -> String {
        String::from("A simple Progressbar")
    }

    fn update(&mut self, message: Message) {
        match message {
            Message::SliderChanged(x) => self.value = x,
        }
    }

    fn view(&self) -> Element<Message> {
        column![
            progress_bar(0.0..=100.0, self.value),
            slider(0.0..=100.0, self.value, Message::SliderChanged).step(0.01)
        ]
        .padding(20)
        .into()
    }
}

```

Please prioritize this as this is blocking our product release. 

### What is the expected behavior?

The functionality should not break and maximize button should be disabled when `resizable` is set to `false`. 

### Version

master

### Operating System

Windows

### Do you have any log output?

```shell
Shared the code snippet and the screenshots above.
```

**Comment by hecrj** [comment]
I have opened #2124 which should fix the issue. Can you give it a try and let me know? Thanks!

**Comment by mayankjoshii** [comment]
Hi @hecrj, 

yes this fix is working. I have attached the success case screenshot with iced 0.10.0 

![image](https://github.com/iced-rs/iced/assets/98211866/3eeb53a6-8a40-4880-9a17-b512664f2323)


Can you please commit and merge this change to the iced 0.10.0 branch, as our project will be referring to that branch and downloading the code at build time using cargo?

**Comment by mayankjoshii** [comment]
Hi @hecrj

This fix is yet not integrated in the iced 0.10.0 version. Can we expect this fix to be added to the iced 10 version and we are dependent on that particular version and choosing some other branch will cause retest of entire GUI.

If updating the version 0.10.0 is not possible then how can we fetch the master version at runtime as we can’t carry the iced code instead we are downloading crates at runtime.

**Comment by hecrj** [comment]
@mayankjoshii I just backported the fix and released `iced_winit 0.10.1`. You should be able to pull the fix with `cargo update`.

Hope that works for you!
